src/main/java/hudson/plugins/s3/callable/S3GzipCallable.java

```diff
@@ -58,17 +66,17 @@ public void progressChanged(ProgressEvent event) {
     }
 
     @Override
+    @SuppressFBWarnings("RV_RETURN_VALUE_IGNORED_BAD_PRACTICE")
     public String invoke(FilePath file) throws IOException, InterruptedException {
         final File localFile = gzipFile(file);
         Upload upload = null;
 
-        try {
-            final InputStream gzipedStream = new FileInputStream(localFile);
+        try (final InputStream gzippedStream = new FileInputStream(localFile)) {
             final ObjectMetadata metadata = buildMetadata(file);
             metadata.setContentEncoding("gzip");
             metadata.setContentLength(localFile.length());
 
-            upload = Uploads.getInstance().startUploading(getTransferManager(), file, gzipedStream, getDest().bucketName, getDest().objectName, metadata);
+            upload = Uploads.getInstance().startUploading(getTransferManager(), file, gzippedStream, getDest().bucketName, getDest().objectName, metadata);
```

Comment:
@varyvol @alecharp 

These select changes inadvertedly broke uploading using a gzipped stream, as the stream is automatically closed before the asynchronous upload task can complete, and causes JENKINS-60916.

This is being fixed in #127, so a merge + release would be appreciated. Currently, uploads using the latest release of the S3 plugin are quite a bit more expensive in bandwidth and S3-storage than they could be.